packages/@n8n/api-types/src/chat-hub.ts

```diff
@@ -107,9 +123,134 @@ export const emptyChatModelsResponse: ChatModelsResponse = {
 	anthropic: { models: [] },
 	google: { models: [] },
 	n8n: { models: [] },
+	// eslint-disable-next-line @typescript-eslint/naming-convention
 	'custom-agent': { models: [] },
 };
 
+export const NodeConnectionTypeSchema: z.ZodType<NodeConnectionType> = z.enum([
+	'ai_agent',
+	'ai_chain',
+	'ai_document',
+	'ai_embedding',
+	'ai_languageModel',
+	'ai_memory',
+	'ai_outputParser',
+	'ai_retriever',
+	'ai_reranker',
+	'ai_textSplitter',
+	'ai_tool',
+	'ai_vectorStore',
+	'main',
+]);
+
+export const OnErrorSchema: z.ZodType<OnError> = z.enum([
+	'continueErrorOutput',
+	'continueRegularOutput',
+	'stopWorkflow',
+]);
+
+export const INodeCredentialsDetailsSchema: z.ZodType<INodeCredentialsDetails> = z.object({
+	id: z.string().nullable(),
+	name: z.string(),
+});
+
+export const INodeCredentialsSchema: z.ZodType<INodeCredentials> = z.record(
+	z.string(),
+	INodeCredentialsDetailsSchema,
+);
+
+export const INodeParameterResourceLocatorSchema: z.ZodType<
+	INodeParameterResourceLocator,
+	z.ZodTypeDef,
+	Omit<INodeParameterResourceLocator, 'value'> & {
+		value?: string | number | null | undefined;
+	}
+> = z
+	.object({
+		__rl: z.literal(true),
+		mode: z.string(),
+		value: z.union([z.string(), z.number(), z.null()]).optional(),
+		cachedResultName: z.string().optional(),
+		cachedResultUrl: z.string().optional(),
+		__regex: z.string().optional(),
+	})
+	.strict()
+	.transform((obj) => {
+		// Zod doesn't like optional undefined...
+		if (!Object.prototype.hasOwnProperty.call(obj, 'value')) {
+			return { ...obj, value: undefined };
+		}
+		return obj as INodeParameterResourceLocator;
+	});
+
+const NodeParamPrimitiveSchema = z.union([
+	z.string(),
+	z.number(),
+	z.boolean(),
+	z.null(),
+	z.undefined(),
+]);
+
+const ResourceMapperValueSchema = z.any() as unknown as z.ZodType<ResourceMapperValue>;
+const FilterValueSchema = z.any() as unknown as z.ZodType<FilterValue>;
+const AssignmentCollectionValueSchema = z.any() as unknown as z.ZodType<AssignmentCollectionValue>;
+
+export const NodeParameterValueSchema: z.ZodType<NodeParameterValueType> = z.lazy(() =>
+	z.union([
+		NodeParamPrimitiveSchema,
+		INodeParameterResourceLocatorSchema,
+		ResourceMapperValueSchema,
```

Comment:
Since `ResourceMapperValueSchema` is defined as `z.any()` at runtime, does this mean `NodeParameterValueSchema` will also accept anything? I guess we can omit these from union until we add tools that has such parameters.